lib/wallet.js

```diff
@@ -15,17 +16,55 @@ const _clientFund = new WeakMap();
  * A class for performing various operations on a wallet.
  * @alias module:nahmii-sdk
  */
-class Wallet extends ethers.Wallet {
+class Wallet extends ethers.Signer {
     /**
-     * Create a Wallet
-     * @param {string} privateKey - The private key for the wallet
+     * Create a Wallet from either a private key or custom signing functions
+     * @param {(string|object)} signer - A private key, or information required for the wallet to have signing capabilities
      * @param {NahmiiProvider} provider - A NahmiiProvider instance
      */
-    constructor(privateKey, provider) {
-        super(privateKey, provider);
+    constructor(signer, provider) {
+        super();
+        this.provider = provider;
+
+        // console.log(this.initializeFromCustomParams(...signer));
+        if (typeof signer === 'string') 
+            this.initializeFromPrivateKey(signer);
+        else 
+            this.initializeFromCustomParams(signer.signMessage, signer.signTransaction, signer.address);
+
         _clientFund.set(this, new ClientFundContract(this));
     }
 
+    /**
+     * Creates signing methods and derives an address from a private key and assigns them to this Wallet instance
+     * @param {string} privateKey - An Ethereum hex encoded private key
+     */
+    initializeFromPrivateKey(privateKey) {
+        const ethersWallet = new ethers.Wallet(privateKey, this.provider);
+        this.signingKey = new SigningKey(privateKey);
```

Comment:
Dont create public properties and/or methods like this. It makes it impossible to document or have a clear API for the users!

Reply:
This is the property the ethers Wallet uses, it's required by internal ethers software Wallet methods. Let me know if you have any suggestions on how else to implement it